### PR TITLE
PR to add status to VerrazzanoProject

### PR DESCRIPTION
**Pull request by desagar**
# Description

A simple status section for per-cluster and overall status for VerrazzanoProject

Fixes VZ-2377

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go [hunk 1]
```diff
@@ -55,8 +58,18 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return result, client.IgnoreNotFound(err)
 	}
 
-	err = r.createOrUpdateNamespaces(ctx, vp, logger)
-	return result, err
+	opResult := controllerutil.OperationResultNone
+	if vp.Namespace == constants.VerrazzanoMultiClusterNamespace {
+		err = r.createOrUpdateNamespaces(ctx, vp, logger)
+		if err != nil {
```
Comment (desagar) [inline application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go:64]: == nil

File: application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go [hunk 2]
```diff
@@ -55,8 +58,18 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return result, client.IgnoreNotFound(err)
 	}
 
-	err = r.createOrUpdateNamespaces(ctx, vp, logger)
-	return result, err
+	opResult := controllerutil.OperationResultNone
+	if vp.Namespace == constants.VerrazzanoMultiClusterNamespace {
+		err = r.createOrUpdateNamespaces(ctx, vp, logger)
+		if err == nil {
```
Comment (jmaron99) [inline application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go:64]: Looks like we ignore any errors? Is that correct?

Perhaps this should be a check for err and return it and then the opResult below that block?

Comment (desagar) [reply application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go:64]: we don't ignore errors - we pass the err to the updateStatus function below, and that results in a status update to failure

Comment (jmaron99) [reply application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go:64]: Never mind - looks like the err does get set and leveraged below in the status update.  Though an "err == nil" seem un-go-like ;)

Comment (desagar) [reply application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go:64]: The opResult does not get used in the status update if err is non-nil

**Review by michael-cico — DISMISSED** [review]

File: application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go (hunk 2 above)
Comment (jmaron99) [reply application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go:64]: Looks like our comments crossed on the interwebs... :)

**Review by mgianatagh — APPROVED** [review]

File: application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go [hunk 3]
```diff
@@ -55,8 +57,14 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return result, client.IgnoreNotFound(err)
 	}
 
+	// Use OperationResultCreated by default since we don't really know what happened to individual NS
+	opResult := controllerutil.OperationResultCreated
 	err = r.createOrUpdateNamespaces(ctx, vp, logger)
-	return result, err
+	if err != nil {
```
Comment (pfmackin) [inline application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go:63]: i think you were going to change this to err == nil

Comment (desagar) [reply application-operator/controllers/clusters/verrazzanoproject/verrazzanoproject_controller.go:63]: I made a minor logic change in response to @mgianatagh 's comment, which changed the sense of the comparison from err == nil to err != nil (which Jon suggested was more go-like anyway)

**Review by pfmackin — APPROVED** [review]

**Pull request merged**